core/src/wallet/common/OperationQuery.cpp

```diff
@@ -110,6 +111,41 @@ namespace ledger {
             return shared_from_this();
         }
 
+        void OperationQuery::count(
+            const std::shared_ptr<api::OperationCountListCallback> &callback) {
+          count().callback(_mainContext, callback);
+        }
+
+        Future<std::vector<api::OperationCount>>
+        OperationQuery::count() {
+          auto self = shared_from_this();
+          return async<std::vector<api::OperationCount>>([=] () {
+            std::vector<api::OperationCount> out;
+            self->performCount(out);
+            return out;
+          });
+        }
+
+        soci::rowset<soci::row> OperationQuery::performCount(soci::session &sql) {
+          return _builder.select("o.type, count(*)")
+              .from("operations").to("o")
+              .outerJoin("blocks AS b", "o.block_uid = b.uid")
+              .groupBy("o.type")
+              .execute(sql);
+        }
+
+        void OperationQuery::performCount(std::vector<api::OperationCount> &operations) {
+          soci::session sql(_pool->getPool());
+          soci::rowset<soci::row> rows = performCount(sql);
```

Comment:
I know that the repo is not representative of this good practice but let's start and keep defining variables `const` by default ;)  (here `rows`, `row`, `type` and `count` are good candidates)

Reply:
done